sippy-ng/src/prow_job_runs/ProwJobRun.js

```diff
@@ -547,151 +556,101 @@ function groupIntervals(filteredIntervals) {
 }
 
 function isOperatorAvailable(eventInterval) {
-  if (
-    eventInterval.locator.includes('clusteroperator/') &&
-    eventInterval.message.includes('condition/Available') &&
-    eventInterval.message.includes('status/False')
-  ) {
-    return true
-  }
-  return false
+  return (
+    eventInterval.tempStructuredLocator.type === 'ClusterOperator' &&
+    eventInterval.tempStructuredMessage.annotations['condition'] ===
+      'Available' &&
+    eventInterval.tempStructuredMessage.annotations['status'] === 'False'
+  )
 }
 
 function isOperatorDegraded(eventInterval) {
-  if (
-    eventInterval.locator.includes('clusteroperator/') &&
-    eventInterval.message.includes('condition/Degraded') &&
-    eventInterval.message.includes('status/True')
-  ) {
-    return true
-  }
-  return false
+  return (
+    eventInterval.tempStructuredLocator.type === 'ClusterOperator' &&
+    eventInterval.tempStructuredMessage.annotations['condition'] ===
+      'Degraded' &&
+    eventInterval.tempStructuredMessage.annotations['status'] === 'True'
+  )
 }
 
 function isOperatorProgressing(eventInterval) {
-  if (
-    eventInterval.locator.includes('clusteroperator/') &&
-    eventInterval.message.includes('condition/Progressing') &&
-    eventInterval.message.includes('status/True')
-  ) {
-    return true
-  }
-  return false
+  return (
+    eventInterval.tempStructuredLocator.type === 'ClusterOperator' &&
+    eventInterval.tempStructuredMessage.annotations['condition'] ===
+      'Progressing' &&
+    eventInterval.tempStructuredMessage.annotations['status'] === 'True'
+  )
 }
 
 // When an interval in the openshift-etcd namespace had a reason of LeaderFound, LeaderLost,
 // LeaderElected, or LeaderMissing, tempSource was set to 'EtcdLeadership'.
 function isEtcdLeadership(eventInterval) {
-  if (eventInterval.tempSource == 'EtcdLeadership') {
-    return true
-  }
-  return false
+  return eventInterval.tempSource === 'EtcdLeadership'
 }
 
 function isPodLog(eventInterval) {
-  if (isEtcdLeadership(eventInterval)) {
-    return false
-  }
-  if (eventInterval.locator.includes('src/podLog')) {
+  if (eventInterval.tempSource === 'PodLog') {
     return true
   }
-  return false
+  return eventInterval.tempSource === 'EtcdLog'
```

Comment:
Just a sanity check: this says that you want to include etcdLog intervals in the `pod-logs` section.  Is that what you want?

Reply:
yes that was intended for now. If we get to the end game every Source gets it's own section, so that would change then I think.